### ISSUE TITLE
Move to upstream IPMI BT driver

### ISSUE THREAD

**Issue by shenki** [openbmc/openbmc#694]
Linux 4.9-rc2 contains our upstream IPMI BT driver.

We should move to this driver, which changes the location of the character device node. Cedric has patches for this.

We should move to matching on the device with udev instead of hardcoding, if that is appropriate.

**Comment by shenki** [comment]
- [ ] Revert the old driver and commit the new
- [ ] Update the device tree
- [ ] Move userspace to the new API

The last step should include the bumping of the kernel so we do not break the build.

**Comment by legoater** [comment]
you will find a backport of the mainline kernel patches and an update of the device tree here : 

  https://github.com/legoater/linux/commits/openbmc-4.7-ipmi-bt

and an updated version of btbridged for the new kernel : 

https://github.com/legoater/btbridge/commits/master

The ```btbridged``` tree also includes an extension adding a ```--device <FILE>``` command line option which can be used by to start the daemon with a custom device file name, one set up by a udev rule for instance.

**Comment by adamliyi** [comment]
Hi @legoater , @shenki asked me to continue work on this issue, will you send your bt driver to mailing list for review? Also the btbridge patch to Gerrit? Or do you want me to do that?

**Comment by legoater** [comment]
@adamliyi: the driver is merged in mainline linux. We should  be fine on that side.

I can send the brbridge patches on the mailing list but  I don't think this required if we go through gerrit. 

What I think we would need to do now, is to integrate all these changes in an openbmc patch and maybe add a udev rule to set the bt device name also. Then we can check the whole on gerrit.

**Comment by adamliyi** [comment]
Submitted the btbridge patch to Gerrit: https://gerrit.openbmc-project.xyz/#/c/904/
https://gerrit.openbmc-project.xyz/#/c/903/, for review.
Tested on Palmetto, the host CPU boots well.

**Comment by williamspatrick** [comment]
@shenki - Is the kernel stuff in already?  Do we need a prereq on a commit with this btbridge stuff I just merged?

**Comment by shenki** [comment]
There's been some confusion here. 

This patch should not have been merged: https://gerrit.openbmc-project.xyz/#/c/904/2

I would have preferred to have seen this one merged only: https://gerrit.openbmc-project.xyz/#/c/903/

As I said in the second comment, we needed to move to the new chardev path at the same time as including the new kernel version.

As this was a change that would cause an incompatible break between userspace and the kernel, I was holding off merging it until we could inform everyone and make the change. However, we now have the userspace changed without the kernel bits in.

I will send a patch to bump the kernel version.
